scripts/compress_images.py

```diff
@@ -0,0 +1,246 @@
+    OUTPUT_DIR = 'compressed_images'
+
+    def __init__(
+        self,
+        input_path: Union[str, pathlib.Path],
+        output_dir: str = OUTPUT_DIR,
+        tolerance: float = TOLERANCE
+    ):
+        """Initialize the ImageCompressor.
+
+        Args:
+            input_path: str. Images directory.
+            output_dir: str. Directory to store compressed images.
+            tolerance: float. Compression threshold.
+        """
+        self.input_path = pathlib.Path(input_path)
+        self.output_dir = output_dir
+        self.tolerance = tolerance
+        self.logger = logging.getLogger(__name__)
+
+    def get_compression_type(self, file_extension: str) -> str:
+        """Determine compression type based on file extension.
+
+        Args:
+            file_extension: str. The file extension of the image.
+
+        Returns:
+            type: str. Compression type (Zip or LZW).
+        """
+        return (
+            'Zip' if file_extension in 
+            self.IMAGE_EXTENSIONS_SUPPORTING_ZIP_COMPRESSION
+            else 'LZW'
+        )
+
+    def run_compression_logic(
+            self,
+            file_extension: str,
+            file_path: pathlib.Path,
+            output_file_path: pathlib.Path,
+            actual_compression: bool
+    ) -> List[CompressedImageInfo] | None:
+        """Helper function for running compression logic."""
+        result_image: List[CompressedImageInfo] = []
+        with Image.open(file_path):
+
+            compression_type = self.get_compression_type(file_extension)
+
+            cmd = [
+                'gm', 'convert',
+                str(file_path),
+                '-strip',
+                '-compress', compression_type,
+                str(output_file_path)
+            ]
+
+            result = subprocess.run(
+                cmd, capture_output=True, text=True, check=True
+            )
+            if result.returncode == 0 and output_file_path.exists():
+                if not actual_compression:
+                    original_size = file_path.stat().st_size
+                    new_size = output_file_path.stat().st_size
+
+                    if new_size < original_size * self.tolerance:
+                        result_image.append({
+                            'path': file_path,
+                            'original_size': original_size,
+                            'new_size': new_size
+                        })
+            else:
+                logging.error(
+                    '[ERROR]: %s occurred on file %s',
+                        file_path,
+                        result.stderr
+                )
+        if actual_compression:
+            return None
+        else:
+            return result_image
+
+    def find_compressible_images(self) -> List[CompressedImageInfo]:
+        """Find images that can be compressed further.
+
+        Returns:
+            result_images: List[CompressedImageInfo]. Compressible images list.
```

Comment:
please fix the docstring type annotations. See previous comment: https://github.com/oppia/oppia/pull/21694#discussion_r1989308028

Reply:
Is this fine @U8NWXD ? 

```suggestion
            result_images: List[CompressedImageInfo]. List of all compressible
                images over repository with attributes as path, original_size
                and new_size.
```

Reply:
No the issue is the type annotation in the docstring, not the description. The annotation should be `list(CompressedImageInfo)`. This convention of using `list(...)` was specified in the comment I linked to above (https://github.com/oppia/oppia/pull/21694#discussion_r1989308028).

Did you have trouble applying the information in the comment I linked to this docstring type annotation? I'm trying to understand why this was confusing so I can help you understand how to address comments like this effectively. Understanding and addressing reviewer comments efficiently is key to open source development, but there have been a number of instances in this PR where comments weren't fully addressed the first time, which drags out the review process

Reply:
Replied to the latest suggestion.